src/main/groovy/org/labkey/gradle/util/BuildUtils.groovy

```diff
@@ -548,6 +549,37 @@ class BuildUtils
             }
         distributionProject.logger.info("${distributionProject.path}: adding ${depProjectPath} as dependency for config ${config}")
         addLabKeyDependency(project: distributionProject, config: config, depProjectPath: depProjectPath, depProjectConfig: "published", depExtension: "module", depVersion: distributionProject.labkeyVersion)
+        Set<String> pathsAdded = new HashSet<>();
+        addTransitiveModuleDependencies(distributionProject, distributionProject.findProject(depProjectPath), config, pathsAdded)
+    }
+
+    private static void addTransitiveModuleDependencies(Project distributionProject, Project depProject, String config, Set<String> pathsAdded)
+    {
+        if (depProject == null)
+            return
+
+        distributionProject.evaluationDependsOn(depProject.getPath())
```

Comment:
Will this work for modules that aren't currently enlisted?
Presuming that it would not, what if we added a new task that verified that no distributions are missing any depended on modules. This wouldn't make distribution creation any easier per-se but it would give us something to run on TeamCity to validate distribution PRs so that we wouldn't produce any invalid distributions.

Also, would this pull in transitive dependencies for projects that aren't intending to produce a functional distribution? e.g. when `makeDistribution = false` in `ModuleDistribution` task

Reply:
Answering my own question: yes, `discvr_modules` distribution includes a bunch of extra modules.

Reply:
This should also work for modules that are not currently enlisted. The published pom file for a module includes all the modules it depends on, so they should get pulled in.  It could be that I'm missing a check for the existence of the project in the recursive case, but I think this generally works.  

We can certainly not do this adding of transitive dependencies if `makeDistribution = false`. Does that seem like the right thing to do? (Wasn't aware of/had forgotten that these non-distribution distributions existed.)

Reply:
Also verified that this pulls in transitive module dependencies for modules that are not enlisted.

Reply:
WRT `makeDistribution = false`, I'm not sure what it would mean to skip transitive dependencies when not making a distribution. Fairly sure we'd want distributions that inherit from these "non-distribution distributions" to inherit transitive dependencies as well.

Reply:
non-distribution distributions are a different matter. Those are distributions without a `ModuleDistribution` task (e.g. `base`). The `makeDistribution` flag is to just zip up a collection of modules.